### PR TITLE
webhook secret documentation

### PR DESCRIPTION
**Pull request by kaloramik**
Screenshot 2024-02-12 at 4 13 20 PM
Screenshot 2024-02-12 at 4 13 27 PM

Screenshot 2024-02-12 at 4 16 38 PM

### CODE REVIEW

File: api-reference/webhooks/webhooks.mdx [hunk 1]
```diff
@@ -3,13 +3,50 @@ title: "Webhooks"
...
+
+Extend will sign each webhook request using a secret unique to the webhook. You can use this signature along with the timestamp to verify that the request is coming from Extend as well as protect against replay attacks.
+
+### Steps to verify a webhook request from Extend
+
+At a high level, the steps to verify a webhook request from Extend are
+1. Retrieve the timestamp of the request from `x-extend-request-timestamp`, the body of the request, and the signing secret associated with the webhook.
+2. Concatentate the timestamp and request body using the following format  
+`v0:${timestamp}:${requestBody}`
+3. Compute a HMAC 256 digest on the resulting string. using the signing secret as the key. The secret can be found in the webhooks table under the "Developer" tab
```
Comment (jordan-extend) [inline api-reference/webhooks/webhooks.mdx:23]:
```suggestion
3. Compute a HMAC 256 digest on the resulting string using the signing secret as the key. The secret can be found in the webhooks table under the "Developer" tab
```

File: api-reference/webhooks/webhooks.mdx [hunk 2]
```diff
@@ -3,13 +3,50 @@ title: "Webhooks"
...
+
+![](/images/webhooks/webhook_secret.png)
+4. Compare this digest with the signature provided in `x-extend-request-signature`. If they are equal, then the request is verified to be from Extend.
+
+here is pseudocode in Javascript to perform the verification
```
Comment (jordan-extend) [inline api-reference/webhooks/webhooks.mdx:28]:
```suggestion
Here is pseudocode in Javascript to perform the verification:
```

File: api-reference/webhooks/webhooks.mdx [hunk 3]
```diff
@@ -3,13 +3,50 @@ title: "Webhooks"
...
+
+## Verifying webhook requests from Extend
+
+Extend will sign each webhook request using a secret unique to the webhook. You can use this signature along with the timestamp to verify that the request is coming from Extend as well as protect against replay attacks.
+
+### Steps to verify a webhook request from Extend
+
+At a high level, the steps to verify a webhook request from Extend are
+1. Retrieve the timestamp of the request from `x-extend-request-timestamp`, the body of the request, and the signing secret associated with the webhook.
+2. Concatentate the timestamp and request body using the following format
```
Comment (jordan-extend) [inline api-reference/webhooks/webhooks.mdx:21]:
```suggestion
2. Concatentate the timestamp and request body using the following format: 
```

File: images/webhooks/.DS_Store
Comment (ebadgio) [inline images/webhooks/.DS_Store:1]: @kaloramik looks like this got added by mistake

**Review by ebadgio — APPROVED** [review]
lgtm, just one flag

**Pull request merged**